Apim 2640 improve console tests

jourdiw opened the pull request:
## Issue

https://gravitee.atlassian.net/browse/APIM-2640

## Description

Try to reduce the time spent on console tests by... : 
- Not using unnecessary `setupAngularJsTesting` in `ngx-service` -- about 5-7 seconds
- Remove unnecessary detect changes 
- Simplify tests to not "double check" a view (i.e. Step 4 harness in v4 API creation workflow) -- 200+ms
- Get rid of checking a view after save (!!). This is unnecessary because we test the body of the post for the save and have other unit tests for the rendering of certain infos. -- 3+ seconds
- Fix flaky test for policy studio test -- 3+seconds * 2
- Deprecate gv widget stat configuration test because it is _very_ slow and will be migrated :) -- 70+s

No loss of code coverage -- Stays at 55.8% coverage.
Lint & Test UI Console -- about 13m30 from 15m+ before

## Additional context

<!-- Add any other context about the PR here -->
<!-- It can be links to other PRs or docs or drawing -->
<!-- Or reproduction steps in case of bug fix -->

<!-- Storybook placeholder -->
---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-tdmrzwxvzy.chromatic.com)
<!-- Storybook placeholder end -->

jourdiw committed:
chore(console): remove unnecessary setupAngularJsTesting

jourdiw committed:
chore(console): remove fixture detect changes

jourdiw committed:
refactor(console): simplify creation v4 api step 4 harness

jourdiw committed:
test(console): get rid of updated check

jourdiw committed:
refactor(console): replace spy with http testing controller

jourdiw committed:
fix(console): stabilize flaky test

jourdiw committed:
chore(console): deprecate slow old tests

gaetanmaisse commented on gravitee-apim-console-webui/src/management/api/creation-v4/api-creation-v4.component.spec.ts:
```diff
@@ -1274,6 +1273,7 @@ describe('ApiCreationV4Component', () => {
   describe('step5', () => {
     const API_ID = 'my-api';
     const PLAN_ID = 'my-plan';
+    let step6Harness: Step5SummaryHarness;
```
Strange to have a `Step5SummaryHarness` instance named `step6Harness` 🤔

phiz71 replied:
We used to have a "documentation Step". When we removed it, the Step6 object became the Step5 object but some variables have not been updated.

jourdiw replied:
It's because the Documentation step is no longer in the project 🙃 It was named when Documentation was Step 5 and the summary was step 6

gaetanmaisse reviewed (commented):
Would like to have a second opinion on `refactor(console): get rid of updated check` (and I think the scope of the commit should be `test` as we are changing only test files)

jourdiw commented:
The commit for getting rid of the updated check is to truly avoid end-to-end testing that does not add any additional coverage and takes more time to load.

There are many tests that followed this structure: 
- change input values of a form
- save the form
- verify that the body of the request to the backend was valid and contained changed input
- verify that the backend response is shown correctly. 

However, this last step of showing the backend response did not add anything to code coverage and took time to load the new view (about 3.5 seconds every time). Essentially, we were not testing the backend response, but rather testing how the input of a component changes how information is shown. 

In the tests where I deleted this last step, I made sure that tests existed already that verified the information is displayed correctly in the DOM. If they didn't exist, I added tests to cover those cases.

Our previous strategies for testing have been rather end-to-end for our console tests. Those types of tests can have their value (such as with the Api Creation Workflow v4), but they can be unnecessary given what we are testing. Especially in the case of dialogs, we want to know that the right calls to the backend have been made to save the information and the right calls have been made to get the new information. But we don't need to know that the DOM has changed. 

cc: @gaetanmaisse

ThibaudAV commented on gravitee-apim-console-webui/src/management/messages/messages.component.spec.ts:
```diff
@@ -51,7 +52,6 @@ describe('MigratedMessagesComponent', () => {
           },
         },
         { provide: RoleService, useValue: { list: () => of(fakeRoles) } },
-        { provide: MessageService, useValue: { sendFromPortal: () => of(1), sendFromApi: () => of(2) } },
```
+1
